deploy/helm/events-collector-config.yaml

```diff
@@ -234,6 +235,7 @@ receivers:
   "persistentvolumes"
   "persistentvolumeclaims"
   "ingresses"
+  "virtualservices"
```

Comment:
Hmm, judging by the source code of the `k8sobjects` receiver, we can list here only objects that we know for sure have definitions in the cluster.
That is OK for the standard resource types, but it is a problem for CRDs, like istio's VirtualService`.

Reply:
I've just added an additional conditional check, that looks for the istio API definition.
`{{- if .Capabilities.APIVersions.Has "networking.istio.io/v1alpha3" -}}`
I have tested that this works with/without istio installed, but it ignores the case where the user has istio api but doesn't know virtualservices. I have not found a way to check for a certain CRD without crashing, so this was the best way I found.

Reply:
`k8sobjects` gets its object data straight from kubernetes server api, so it should theoretically know everything I can search for with kubectl, i think.

Reply:
The issu is that the function only works with a specific api version, and doesn't support wildcards, so we'd have to maintain it with new istio API versions..

Reply:
There are two disadvantages of the `.Capabilities.APIVersions.Has` approach:
* It's not compatible with EKS addons. The EKS addon will never monitor `VirtualService` this way.
* The check is only applied during installation or update of the Helm chart. If Istio is added to the cluster after `k8s collector` is deployed, it won't notice it.

Reply:
Can you expand on the issue of the EKS addon?

The second point could theoretically be solved with having some kind of Mutating WebHook or Kubernetes Operator that listens to changes in the cluster, though I don't have experience with using it, and it sounds like too much for this issue?
We could also just notify the user to update the chart if we find istio metrics in a cluster that isn't configured to scrape `VirtualService`, or similarly introduce a flag to values for this, which again, requires user action.

I'm currently not aware of other approaches, sadly.